### PR TITLE
[58_7] new CI for archlinux

### PR DESCRIPTION
**Pull request by Oyyko**
## Why you open this Pull Request?
Add Build CI for Archlinux

### CODE REVIEW

**Comment by Oyyko** [comment]
Runs well.

File: .github/workflows/ci-archlinux.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    container: archlinux:base-devel
+    runs-on: ${{ matrix.os }}
+    concurrency:
+        group: ${{ github.ref }}-${{ github.base_ref }}-${{ github.head_ref }}-Archlinux-${{ matrix.kind }}
+        cancel-in-progress: true
+    steps:
+      - name: PacMan
+        run: |
+          sudo pacman -Syyu --noconfirm --needed
+          sudo pacman -S --noconfirm --needed qt5-base qt5-svg freetype2 sqlite libpng libiconv zlib libjpeg curl texlive-core python libxext xmake unzip
```
[the remark this answers is not in the document]
Comment (darcy-shen) [reply .github/workflows/ci-archlinux.yml:38]: and use qt6 is encouraged

File: .github/workflows/ci-archlinux.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - uses: actions/checkout@v1
+      - name: Xrepo update
+        run: yes | XMAKE_ROOT=y xrepo update-repo
+      - name: Config
+        run: XMAKE_ROOT=y xmake config -m release --yes
+      - name: Build
+        run: XMAKE_ROOT=y xmake build --yes --verbose --diagnosis --jobs=8 --all
```
Comment (darcy-shen) [inline .github/workflows/ci-archlinux.yml:45]: setting the XMAKE_ROOT env would be better: https://github.com/xmake-io/xmake-repo/blob/dev/.github/workflows/archlinux.yml#L37-L38

File: .github/workflows/ci-archlinux.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (darcy-shen) [reply .github/workflows/ci-archlinux.yml:38]: There are discussions here: https://aur.archlinux.org/packages/mogan

In this pr, just use the package provided by xrepo, and later, we may switch to libraries shipped in ArchLinux.

File: .github/workflows/ci-archlinux.yml [hunk 3]
```diff
@@ -23,23 +23,28 @@ on:
...
   build:
     strategy:
-      matrix:
-        os: [ubuntu-latest]
-        kind: [static, shared]
     container: archlinux:base-devel
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     concurrency:
         group: ${{ github.ref }}-${{ github.base_ref }}-${{ github.head_ref }}-Archlinux-${{ matrix.kind }}
```
Comment (jingkaimori) [inline .github/workflows/ci-archlinux.yml:29]: why not remove `matrix.kind` here?

File: .github/workflows/ci-archlinux.yml (hunk 2 above)
Comment (darcy-shen) [reply .github/workflows/ci-archlinux.yml:50]: XMAKE_ROOT 可以设置为全局变量的 https://docs.github.com/en/actions/learn-github-actions/variables

File: .github/workflows/ci-archlinux.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      - '!src/Plugins/Windows/**'
+      - 'tests/**'
+      - 'misc/xmake/packages.lua'
+      - 'xmake.lua'
+      - '.github/workflows/ci-archlinux.yml'
+jobs:
+  build:
+    strategy:
+    container: archlinux:base-devel
+    runs-on: ubuntu-latest
```
Comment (darcy-shen) [inline .github/workflows/ci-archlinux.yml:27]: ubuntu-latest会变的，后面有出错的可能，建议直接使用 ubuntu-22.04

Comment (darcy-shen) [reply .github/workflows/ci-archlinux.yml:27]: 假设 ubuntu-latest 升级到了 ubuntu-24.04，然后出错了，这样我们就浪费了精力去处理这个问题。

https://zh.moegirl.org.cn/%E8%8A%82%E8%83%BD%E4%B8%BB%E4%B9%89

节能主义

**Comment by Oyyko** [comment]
I will open a new PR.

**Pull request closed without merging**